sql/2019/17_CDN/17_01.sql

```diff
@@ -0,0 +1,17 @@
+#standardSQL
+# 17_01: Top CDNs used on the root HTML pages
+SELECT
+  client,
+  _cdn_provider AS cdn,
+  COUNT(0) AS freq,
+  SUM(COUNT(0)) OVER (PARTITION BY client) AS total,
+  ROUND((COUNT(0) * 100 / SUM(COUNT(0)) OVER (PARTITION BY client)), 2) AS pct
+FROM
+  `httparchive.almanac.requests`
+WHERE
+  firstHtml

```

Comment:
the count of total pages based on 17_02 is 5,297,442 and 4,371,974 for mobile and desktop respectively. yet the 'total' in this query is 5,558,214 and 4,550,580. I don't think the firstHtml query is applying properly to the partition window

Reply:
which is different than 17_02c which has 5,290,857, 4,362,182. This last number appears to be more correct since it both filters for firstHtml and groups by pageid which presumably removes the redirects.